packages/drivers/routerlicious-driver/src/documentService.ts

```diff
@@ -277,9 +282,20 @@ export class DocumentService
 				summarizeProtocolTree: shouldSummarizeProtocolTree,
 			};
 
+			connection.once("disconnect", (reason: IAnyDriverError) => {
+				if (reason.errorType === RouterliciousErrorTypes.authorizationError) {
+					this.refreshTokenOnConnect = true;
+				}
+			});
 			return connection;
 		} catch (error: any) {
-			if (error?.statusCode === 401) {
+			if (
+				typeof error === "object" &&
+				error !== null &&
+				((error as Partial<IR11sError>).errorType ===
+					RouterliciousErrorTypes.authorizationError ||
+					error?.statusCode === 401)
```

Comment:
It's a bit weird to see checking both statusCode and errorType. We usually try to operate only errorTypes once a translation from status codes to error types happened.